packages/core/lib/feature-flag-delegate.js

```diff
@@ -14,7 +14,16 @@ function add (existingFeatures, name, variant) {
     variant = jsonStringify(variant)
   }
 
-  existingFeatures[name] = variant
+  var found = false
+  for (var i = 0; i < existingFeatures.length; i++) {
+    if (existingFeatures[i].name === name) {
+      found = true
+      existingFeatures[i].variant = variant
```

Comment:
Would it be safer to replace rather than mutate the object, since an event.features may also contain it?

e.g.

```suggestion
      existingFeatures[i].variant = { name, variant }
```

Reply:
I don't think it makes any difference here, as the flag will have been _cloned_ from the event, rather than referenced.